### PR TITLE
Prepare v1.0.4 release

### PR DESCRIPTION
**Pull request by azdagron**
This prepares the v1.0.4 release branch by cherry-picking the commits that add the new disable_spiffe_cert_validation functionality from the following PRs:

https://github.com/spiffe/spire/pull/3009
https://github.com/spiffe/spire/pull/3014
https://github.com/spiffe/spire/pull/3020
https://github.com/spiffe/spire/pull/3034

It also bumps the version and updates the CHANGELOG.

### CODE REVIEW

File: test/integration/suites/upgrade/versions.txt [hunk 1]
```diff
@@ -5,3 +5,5 @@
 1.0.0
 1.0.1
 1.0.2
+1.0.3
+1.0.4
```
Comment (amartinezfayo) [inline test/integration/suites/upgrade/versions.txt:9]: 1.0.4 should not be included, since this is 1.0.4.

**Review by amartinezfayo — COMMENTED** [review]
Thanks @azdagron, looks good.
My only comment is with the upgrade integration test.

**Comment by azdagron** [comment]
Thanks, @amartinezfayo . I've amended the previous commit to fix that and force pushed. Hope that is ok. I wanted to retain a very set of commits in the release branch.

File: test/integration/suites/upgrade/versions.txt (hunk 1 above)
Comment (azdagron) [reply test/integration/suites/upgrade/versions.txt]: Woof. Thanks!

**Review by amartinezfayo — APPROVED** [review]

**Review by rturner3 — APPROVED** [review]

**Pull request merged**